primitives/payment-handler/src/tests.rs

```diff
@@ -256,17 +333,47 @@ mod asset_fee_handler {
 			});
 		}
 
+		#[test]
+		fn withdraw_and_deposit_into_treasury_works_with_vouchers() {
+			ExtBuilder::default().vouchers(&[(ALICE, 15)]).build().execute_with(|| {
+				let fee = 15;
+				let alice_balance_before = Assets::balance(WHITELISTED_ASSET_ID, ALICE);
+				let fee_beneficiary = FERDIE;
+
+				let alice_initial_vouchers = VOUCHERS
+					.with_borrow(|voucher_store| voucher_store.get(&ALICE).copied())
+					.expect("Should contain remaining vouchers");
+
+				TestAssetFeeHandler::withdraw_and_deposit_into_treasury(
+					&ALICE,
+					VOUCHER_ASSET_PAYMENT,
+					&fee_beneficiary,
+					fee,
+				)
+				.unwrap();
+
+				assert_eq!(Assets::balance(WHITELISTED_ASSET_ID, ALICE), alice_balance_before);
+				assert_eq!(Assets::balance(WHITELISTED_ASSET_ID, fee_beneficiary), 0);
+
+				// check that the requested vouchers have been deducted from storage
+				let alice_current_vouchers = VOUCHERS
+					.with_borrow(|voucher_store| voucher_store.get(&ALICE).copied())
+					.expect("Should contain remaining vouchers");
+				assert_eq!(alice_current_vouchers, alice_initial_vouchers - fee);
+			});
+		}
```

Comment:
This is actually very interesting. I guess it makes sense the way it is implemented, i.e, successfully deposit, but don't deposit anything. However, instead of `_works`, I would call this test: 

`withdraw_and_deposit_into_treasury_with_vouchers_does_not_err`

And we should add a comment that this is just the defensive behavior, as we don't want an error.